### DISCUSSION TITLE
Question about the child_sa timer？

### DISCUSSION THREAD

**Discussion by yw1573** [strongswan/strongswan#1200] [Q&A] [answered]
Dear：
    1.In ikev1, who will trigger the rekey of child_sa of ike sa, strongswan or linux kernel？
    2.In ikev1, who will trigger the rekey of child_sa of ipsec sa, strongswan or linux kernel？
Whether the timer is in strongswan or linux kernel, sorry, I just touched the code of this piece, it is a bit complicated, so I had to ask for help

**Comment by strongX509** [comment]
The rekeying of a CHILD or IPsec SA is always triggered by a timer in the Linux kernel but executed by strongSwan via IKE messages.

The rekeying of an IKE SA is always triggered by a strongSwan timer.

**Reply by yw1573** [reply]
Thank you very much, I repeat, the rekey of ike sa is triggered by the timer of strongswan, and the rekey of ipsec sa is triggered by the timer of linux kernel

**Comment by strongX509** [comment, marked as answer]
Yes, this is how rekeying works
